pkg/controllers/controller.go

```diff
@@ -173,18 +194,41 @@ func (h *handler) prepareHandler(registrationObj *v1.Registration) SCCHandler {
 	}
 }
 
-func (h *handler) OnSecretChange(name string, incomingObj *corev1.Secret) (*corev1.Secret, error) {
+func (h *handler) OnSecretChange(_ string, incomingObj *corev1.Secret) (*corev1.Secret, error) {
 	if incomingObj == nil || incomingObj.DeletionTimestamp != nil {
 		return incomingObj, nil
 	}
+
 	if h.isRancherEntrypointSecret(incomingObj) {
+		// TODO(alex): sync on this to validate logic
+		// TODO: something to handle adopting new
+		if !shouldManage(incomingObj, h.managedByName) {
```

Comment:
This also needs to apply to registration objects

Reply:
Solid.

So for registration objects I'm thinking it should likely be identical to secrets, even tho it could be slightly different. I could see us handling it different because we know the secret broker creates the registration with the proper label. So in theory the whole `OnRemove` and `OnChange` could happen via a scoped version limiting to just ones with the management label matching.

However that would fully exclude any manually created `registration` objects without the label. Which could be a minor documentations note for any "power users" wishing to not use the secret broker (which I think we agree is the officially supported route).

Reply:
Yep - so this works pretty well IMO...I have updated it to use scoped `OnRemove` and `OnChange`. This way only the ones created by us will be managed by us, OR if someone manually creates one with the correct label the operator will try (and fail).

That second one in mind, I added to the ADR that only the `secret-broker` method is supported for now and that later patching will refactor the dependency for `secret-broker` out. At which time we _could_ start allowing end-users to directly create `Registration` objects.